### ISSUE TITLE
Rule: newline-per-method [$50]

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#4538]
Require that each method is on a separate line.

The following patterns are considered problems:

```js
_.chain({}).map(foo).filter(bar).value();
```

The following patterns are not considered problems:

```js
_
    .chain({})
    .map(foo)
    .filter(bar)
    .value();
```

**Comment by eslintbot** [comment]
@gajus Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
@gajus I think this will be too difficult to implement in a way that won't clash with `indent`. I'd suggest trying to write it yourself and let us know how it works when `indent` is enabled.

**Comment by gajus** [comment]
@nzakas How would it clash with `indent` rule?

The reason I am asking is because I am using `indent` rule and the latter example is how I am structuring all my code already. It is just not enforced.

**Comment by nzakas** [comment]
As I said, please try writing it yourself and see what happens. It might be okay, but it might not.

**Comment by gajus** [comment]
@nzakas cool bot.

**Comment by nzakas** [comment]
Bountysource does it when a bounty is applied. Does this mean you aren't going to try writing this yourself? (This issue isn't accepted at this point, which means we aren't committed to doing it.)

**Comment by gajus** [comment]
Thats fine. I opened the bounty to motivate someone else try it who needs it as much as I do. Unfortunately, I am over the capacity.

**Comment by nzakas** [comment]
We honestly just don't have enough people to keep up with the demand for new rules, that's why I ask people to create custom rule instead.

I'm still not sure this rule is a good idea, so if you're not going to try it yourself, then I need to close the issue.

**Comment by rpatil26** [comment]
@gajus - Is this what you're looking for? The code is not complete. I need to add more cases to confirm if all the scenarios are covered. Just thought to create a PR in advance to check.

@nzakas I know this issue is still is proposal stage and I am not even sure if we'll accept this, but I thought to have stab at it anyways. Do you think we can accept this rule and go ahead? If so, I'll need to finalize and polish it further.

**Comment by nzakas** [comment]
@rpatil26 if you're willing to write it, then go for it. I'll mark the issue as accepted.

**Comment by rpatil26** [comment]
Thanks @nzakas. I will resume work on it. I expect to complete it over the weekend.

**Comment by nzakas** [comment]
Great, thanks!

**Comment by rpatil26** [comment]
@nzakas and @gajus Updated PR with changes (configuration depth to ignore). Request you have a close look and see if this rule looks fine.
